### PR TITLE
ENG-1253 remove redundant jira client

### PR DESCRIPTION
**Pull request by pashafateev**

### CODE REVIEW

**Review by daabr — COMMENTED** [review]
Also need to update these files:
- `runtimes/pythonrt/py-sdk/pyproject.toml`
- `runtimes/pythonrt/py-sdk/docs/conf.py` (SDK version)
- `runtimes/pythonrt/py-sdk/docs/index.rst`
- `runtimes/pythonrt/py-sdk/pyproject.toml` (lib list + SDK version)
- `runtimes/pythonrt/dev-requirements.txt`
- `runtimes/pythonrt/requirements.txt`

File: runtimes/pythonrt/py-sdk/autokitteh/atlassian.py [hunk 1]
```diff
@@ -5,7 +5,6 @@
 import os
 
 from atlassian import Confluence, Jira
-from jira import JIRA
```
Comment (tebeka) [inline runtimes/pythonrt/py-sdk/autokitteh/atlassian.py:8]: What's the different between `Jira` and `JIRA`? Who added it in the first time and why?

File: runtimes/pythonrt/py-sdk/autokitteh/atlassian.py [hunk 2]
```diff
@@ -15,7 +14,7 @@
 __TOKEN_URL = "https://auth.atlassian.com/oauth/token"
 
 
-def atlassian_jira_client(connection: str, **kwargs) -> Jira:
+def jira_client(connection: str, **kwargs) -> Jira:
```
Comment (tebeka) [inline runtimes/pythonrt/py-sdk/autokitteh/atlassian.py:17]: This is an API breaking change. 
Do we know if there's a current client who's using it?
If so, keep the old function as well, something like:

```python
import warnings

def atlassian_jira_client(connection: str, **kwargs) -> Jira:
    warnings.warn("`attlasian_jira_client` is deprecated, use `jira_client`", DeprecationWarning)
    return jira_client(connection, **kwargs)

```

File: runtimes/pythonrt/py-sdk/autokitteh/atlassian.py [hunk 3]
```diff
@@ -15,7 +14,7 @@
...
     """Initialize an Atlassian Jira client, based on an AutoKitteh connection.
 
     API reference:
```
Comment (tebeka) [inline runtimes/pythonrt/py-sdk/autokitteh/atlassian.py:20]: Missing docstring for the arguments, see the original docstring.

**Review by tebeka — CHANGES_REQUESTED** [review]

File: runtimes/pythonrt/py-sdk/autokitteh/atlassian.py (hunk 2 above)
Comment (pashafateev) [reply runtimes/pythonrt/py-sdk/autokitteh/atlassian.py:17]: @daabr I don't think anyone but us is using this correct?

File: runtimes/pythonrt/py-sdk/autokitteh/atlassian.py (hunk 1 above)
Comment (daabr) [reply runtimes/pythonrt/py-sdk/autokitteh/atlassian.py:8]: I added 2 Atlassian client libs that seemed to be the most popular ones - one only for Jira (client is `jira.JIRA`) and the other is a lib for multiple Atlassian products (specifically Confluence, but also Jira - `atlassian.Jira` and `atlassian.Confluence` respectively).

After experiments and actual usage, we learned that the Jira-only lib is less suited for our auth needs, so this PR removes it in favor of using only the atlassian lib for Jira too, not just Confluence.

File: runtimes/pythonrt/py-sdk/autokitteh/atlassian.py (hunk 2 above)
Comment (daabr) [reply runtimes/pythonrt/py-sdk/autokitteh/atlassian.py:17]: We have one customer who requested the Jira integration, but they don't have any projects that use it yet. The only places that reference this are our projects in the `samples` and `kittehub` repos - and there are PRs to fix them, which will be merged as soon as this PR is merged.

File: runtimes/pythonrt/py-sdk/autokitteh/atlassian.py (hunk 3 above)
Comment (pashafateev) [reply runtimes/pythonrt/py-sdk/autokitteh/atlassian.py:20]: I don't understand what you mean. Missing docstring in `jira_client` ?
Do you mean this?

>    Args:
        connection: AutoKitteh connection name.

File: runtimes/pythonrt/py-sdk/autokitteh/atlassian.py (hunk 1 above)
Comment (tebeka) [reply runtimes/pythonrt/py-sdk/autokitteh/atlassian.py:8]: OK

File: runtimes/pythonrt/py-sdk/autokitteh/atlassian.py (hunk 2 above)
Comment (tebeka) [reply runtimes/pythonrt/py-sdk/autokitteh/atlassian.py:17]: Go ahead then.

File: runtimes/pythonrt/py-sdk/autokitteh/atlassian.py (hunk 3 above)
Comment (tebeka) [reply runtimes/pythonrt/py-sdk/autokitteh/atlassian.py:20]: Ah, github diff clipped this. Please document `kwargs`

**Review by tebeka — APPROVED** [review]

**Pull request merged**